client/verta/tests/unit_tests/strategies.py

```diff
@@ -26,14 +28,33 @@ def int64(draw) -> int:
 
 
 @st.composite
-def artifact_proto(draw) -> CommonService_pb2.Artifact:
-    """Generate a mocked Artifact protobuf object."""
-    key: str = draw(st.text(_VALID_FLAT_KEY_CHARS, min_size=1))
+def attribute_proto(draw) -> CommonService_pb2.KeyValue:
+    """Generate a mocked KeyValue protobuf object representing an attribute."""
+    return CommonService_pb2.KeyValue(
+        key=draw(st.text()),
+        value=python_to_val_proto(
+            draw(json_strategy),
+            allow_collection=True,
+        ),
+    )
+
+
+@st.composite
+def artifact_proto(draw, key: Optional[str] = None) -> CommonService_pb2.Artifact:
+    """Generate a mocked Artifact protobuf object.
+
+    Parameters
+    ----------
+    key : str, optional
+        Key to use for the artifact. If not provided, one will be generated.
+
+    """
+    key: str = draw(st.text(_VALID_FLAT_KEY_CHARS, min_size=1)) if key is None else key
     filename_extension: str = draw(st.text(ascii_lowercase))
 
     return CommonService_pb2.Artifact(
         key=key,
-        path=f"{draw(st.text(ascii_letters + '/'))}/{key}.{filename_extension}",
+        path=f"{draw(st.text(ascii_letters, min_size=1))}/{key}.{filename_extension}",
```

Comment:
This sometimes created a path that started with `/`, which can mess with functionality involving saving artifacts to disk: The client might try to write the artifact to a root `/foo`, which will fail.

Reply:
Nice catch!